apps/vaults/components/RewardsTab.tsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import {useStakingRewards} from '@vaults/contexts/useStakingRewards';
+import * as StakingRewardsActions from '@vaults/utils/actions/stakingRewards';
+import {useTransaction} from '@veYFI/hooks/useTransaction'; // TODO: make a common hook
+import {validateAllowance} from '@veYFI/utils/validations'; // TODO: make it common
+import {Button} from '@yearn-finance/web-lib/components/Button';
+import {useWeb3} from '@yearn-finance/web-lib/contexts/useWeb3';
+import {toAddress} from '@yearn-finance/web-lib/utils/address';
+import {toNormalizedBN} from '@yearn-finance/web-lib/utils/format';
+import {formatCounterValue} from '@yearn-finance/web-lib/utils/format.value';
+import {Input} from '@common/components/Input';
+import {useWallet} from '@common/contexts/useWallet';
+import {useAllowances} from '@common/hooks/useAllowances';
+import {useBalance} from '@common/hooks/useBalance';
+import {approveERC20} from '@common/utils/actions/approveToken';
+
+import type {ReactElement} from 'react';
+import type {TAddress} from '@yearn-finance/web-lib/types';
+import type {TYearnVault} from '@common/types/yearn';
+
+const DISPLAY_DECIMALS = 10;
+const trimAmount = (amount: string | number): string => Number(Number(amount).toFixed(DISPLAY_DECIMALS)).toString(); 
```

Comment:
Move to utils? Add tests? :)

Reply:
All good, one last thing, as the function is taking a string, should we add a check for when `isNaN`?